src/com/unbounce/dogstatsd/core.clj

```diff
@@ -32,46 +33,58 @@
                                tags)))))
 
 (defn increment
-  [metric & {:keys [tags sample-rate]}]
-  (let [tags (str-array tags)]
-    (if sample-rate
-      (.incrementCounter client metric sample-rate tags)
-      (.incrementCounter client metric tags))))
+  ([metric]
+   (.incrementCounter client metric -empty-array))
+  ([metric {:keys [tags sample-rate]}]
+   (let [tags (str-array tags)]
+     (if sample-rate
+       (.incrementCounter client metric sample-rate tags)
+       (.incrementCounter client metric tags)))))
 
 (defn decrement
-  [metric & {:keys [tags sample-rate]}]
-  (let [tags (str-array tags)]
-  (if sample-rate
-    (.decrementCounter client metric sample-rate tags)
-    (.decrementCounter client metric tags))))
+  ([metric]
+   (.decrementCounter client metric -empty-array))
+  ([metric {:keys [tags sample-rate]}]
+   (let [tags (str-array tags)]
+     (if sample-rate
+       (.decrementCounter client metric sample-rate tags)
+       (.decrementCounter client metric tags)))))
 
 (defn gauge
-  [metric value {:keys [sample-rate tags]}]
-  (let [value       (double value)
-        tags        (str-array tags)
-        sample-rate (when sample-rate (double sample-rate))
-        f           (fn [^String metric ^double value {:keys [^double sample-rate #^"[Ljava.lang.String;" tags]}]
-                      (if sample-rate
-                        (.recordGaugeValue client metric value sample-rate tags)
-                        (.recordGaugeValue client metric value tags)))]
-    (f metric value {:sample-rate sample-rate
-                     :tags tags})))
+  ([^String metric ^Double value]
+   (.recordGaugeValue client metric value -empty-array))
+  ([^String metric ^Double value {:keys [sample-rate tags]}]
+   (let [tags        (str-array tags)
+         sample-rate ^Double sample-rate]
+     (if sample-rate
+       (.recordGaugeValue client metric value sample-rate tags)
+       (.recordGaugeValue client metric value tags)))))
```

Comment:
@gzmask I cleaned this API up a bit to remove the need to create an anonymous function

Reply:
Reason of the anonymous function is that I wanted to get rid of the propagate effect of type annotations - so that clojure code remains clojurish and Java types are localized to where the interop happens. Not sure if this is a good practice.

Reply:
I'm uncertain what propogate effects you're thinking of. 

From the outside you won't need to know about the typehints. The typehints are used to help the clojure compiler pick a function rather than use reflection.